aten/src/ATen/native/cpu/BinaryOpsKernel.cpp

```diff
@@ -12,11 +12,17 @@ namespace at { namespace native {
 namespace {
 
 using namespace vec256;
-
+ 
 void add_kernel(TensorIterator& iter, Scalar alpha_scalar) {
   if (iter.dtype() == ScalarType::Bool) {
     auto alpha = alpha_scalar.to<bool>();
     cpu_kernel(iter, [=](bool a, bool b) -> bool { return a + b * alpha; });
+  } else if (isComplexType(iter.dtype())) {
+    AT_DISPATCH_COMPLEX_TYPES_AND(kBFloat16, iter.dtype(), "add_cpu/sub_cpu", [&]() {
```

Comment:
I don't see why you dispatch to kBFloat16 here. That doesn't seem right to me.

Reply:
In an ideal world I guess this would be combined with the if statement above with something like `iter.dtype() == ScalarType::Bool || isComplexType(iter.dtype())` and then `AT_DISPATCH_COMPLEX_TYPES_AND(kBool, ...`

Reply:
(This admonition applies below too)

Reply:
Uggh, sorry I didn't quite get what was going on with the `DISPATCH` macros. I have now cleaned up the code.